byzcoin/transaction.go

```diff
@@ -420,6 +382,45 @@ func (sc StateChange) ShortString() string {
 	return sc.toString(false)
 }
 
+// Key returns the key that should be used in a key/value database.
+func (sc *StateChange) Key() []byte {
```

Comment:
Maybe you can put these methods on StateChange (no *) and then you can remove the copy to []KVPair in StoreAll.

To check the implementation: https://stackoverflow.com/questions/27803654/explanation-of-checking-if-value-implements-interface

Reply:
your suggestion didn't work, I made a simple test here - https://play.golang.org/p/waJ1f-6pz8a

regardless of whether it's a pointer receiver or a regular receiver, the `func addAreas(shapes []Shape) int` won't compile...